### PR TITLE
[Experiment] New WP Dashboard Widgets

### PR DESCRIPTION
**Pull request by arielk**

### CODE REVIEW

**Comment by rodolphebertozzo** [comment]
Base 3.1.0 ?

File: assets/dev/js/admin/admin.js [hunk 1]
```diff
@@ -435,5 +435,11 @@
 		window.elementorAdmin = new ElementorAdmin();
 
 		elementorCommon.elements.$window.trigger( 'elementor/admin/init' );
+
+		$( '.elementor_finder' ).on( 'click', function(e) {
```
Comment (matipojo) [inline assets/dev/js/admin/admin.js:439]:
```suggestion
		$( '.elementor-finder' ).on( 'click', function( e ) {
```

File: core/admin/admin.php [hunk 2]
```diff
@@ -417,7 +417,7 @@ public function elementor_dashboard_overview_widget() {
 			</div>
 			<?php if ( $recently_edited_query->have_posts() ) : ?>
 				<div class="e-overview__recently-edited">
-					<h3 class="e-overview__heading"><?php echo __( 'Recently Edited', 'elementor' ); ?></h3>
+					<h3 class="e-heading e-divider_bottom"><?php echo __( 'Recently Edited', 'elementor' ); ?></h3>
```
Comment (matipojo) [inline core/admin/admin.php:420]: Always use `-` instead of `_` in class names.

File: modules/dashboard-widgets/module.php [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+	 * @var WP_Query $recently_edited_query
+	 */
+	private $recently_edited_query;
+
+	public function get_name() {
+		return 'widgets';
+	}
+
+	public function __construct() {
+
```
Comment (matipojo) [inline modules/dashboard-widgets/module.php:33]: Remove empty line.

File: modules/dashboard-widgets/module.php [hunk 4]
```diff
@@ -0,0 +1,482 @@
...
+		$this->set_vars();
+
+		remove_action( 'welcome_panel', 'wp_welcome_panel' );
+
+		if ( 3 >= $this->recently_edited_query->post_count ) {
+			add_action( 'welcome_panel', [ $this, 'welcome_dashboard_widget_render' ], PHP_INT_MAX );
+		}
+
+		//add_action( 'admin_enqueue_scripts', [ $this, 'load_assets' ] );
```
Comment (matipojo) [inline modules/dashboard-widgets/module.php:42]: Remove.

File: modules/dashboard-widgets/module.php [hunk 5]
```diff
@@ -0,0 +1,482 @@
...
+
+		if ( is_network_admin() ) {
+			add_action( 'wp_network_dashboard_setup', [ $this, 'add_dashboard_widgets' ], 999 );
+		} elseif ( is_user_admin() ) {
+			add_action( 'wp_user_dashboard_setup', [ $this, 'add_dashboard_widgets' ], 999 );
+		} else {
+			add_action( 'wp_dashboard_setup', [ $this, 'add_dashboard_widgets' ], 999 );
```
Comment (matipojo) [inline modules/dashboard-widgets/module.php:49]: Why 999? (remove or explain vie code comment.)

File: modules/dashboard-widgets/module.php [hunk 6]
```diff
@@ -0,0 +1,482 @@
...
+	}
+
+	public function __construct() {
+
+		$this->set_vars();
+
+		remove_action( 'welcome_panel', 'wp_welcome_panel' );
+
+		if ( 3 >= $this->recently_edited_query->post_count ) {
+			add_action( 'welcome_panel', [ $this, 'welcome_dashboard_widget_render' ], PHP_INT_MAX );
```
Comment (matipojo) [inline modules/dashboard-widgets/module.php:39]: Why PHP_INT_MAX? (remove or explain vie code comment.)

File: modules/dashboard-widgets/module.php [hunk 7]
```diff
@@ -0,0 +1,482 @@
...
+			'post_type' => 'any',
+			'post_status' => [ 'publish', 'draft' ],
+			'posts_per_page' => '3',
+			'meta_key' => '_elementor_edit_mode',
+			'meta_value' => 'builder',
+			'orderby' => 'modified',
+		];
+
+		$this->recently_edited_query = new WP_Query( $recently_edited_query_args );
+
```
Comment (matipojo) [inline modules/dashboard-widgets/module.php:64]: Empty line, please follow all lint errors: https://github.com/elementor/elementor/pull/13013/checks?check_run_id=1355195256

File: modules/dashboard-widgets/module.php [hunk 8]
```diff
@@ -0,0 +1,482 @@
...
+		$show_welcome_panel = get_user_meta( get_current_user_id(), 'show_welcome_panel', true );
+		if ( ! $show_welcome_panel ) {
+			$widgets['e-dashboard-widget-videos'] = array(
+				'label' => esc_html__( 'Elementor Video Tutorials', 'elementor' ),
+				'callback' => [ $this, 'dashboard_videos_render' ],
+			);
+		}
+
+		$widget_backup = array();
```
Comment (matipojo) [inline modules/dashboard-widgets/module.php:92]:
```suggestion
		$widget_backup = [];
```

File: modules/dashboard-widgets/module.php [hunk 9]
```diff
@@ -0,0 +1,482 @@
...
+
+		foreach ( $widgets as $widget_id => $widget ) {
```
Comment (matipojo) [inline modules/dashboard-widgets/module.php:94]: Explain the approach.

File: modules/dashboard-widgets/module.php [hunk 10]
```diff
@@ -0,0 +1,482 @@
...
+
+	public function add_global_widget_class( $classes ) {
+		$classes[] = 'e-dashboard-widget';
+
+		return $classes;
+	}
+
+	public function welcome_dashboard_widget_render() {
+		$create_page_url = Utils::get_create_new_post_url();
+		$create_post_url = Utils::get_create_new_post_url( 'post' );
```
Comment (matipojo) [inline modules/dashboard-widgets/module.php:128]:
```suggestion
		$create_post_url = Utils::get_create_new_post_url( 'post' );
```

File: modules/dashboard-widgets/module.php [hunk 11]
```diff
@@ -0,0 +1,482 @@
...
+
+		$action_links = array(
```
Comment (matipojo) [inline modules/dashboard-widgets/module.php:130]: Short arrays everywhere.

**Review by matipojo — CHANGES_REQUESTED** [review]

**Comment by KingYes** [comment]
Please continue there: #13019

**Pull request closed without merging**